.github/instructions/custom_auth_tech.instructions.md

```diff
@@ -0,0 +1,212 @@
+-   Common HTTP operations abstracted in `BaseApiClient`
+-   Consistent request/response handling patterns
+
+### HTTP Standards
+
+-   Use `IHttpClient` interface for all HTTP operations
+-   Proper correlation ID handling in all requests
+-   Consistent header management across requests
+-   Structured error response handling
+
+### Request/Response Types
+
+-   Strongly typed request and response interfaces
+-   Consistent naming conventions (`*Request`, `*Response`)
+-   Proper validation of API responses
+-   Error response standardization
+
+## Testing Standards
+
+### Unit Testing Requirements
+
+-   **Target coverage: >85%**
+-   Tests mirror source code folder structure
+-   Each major class has dedicated test file
+-   Mock dependencies within custom_auth feature only
+-   Use real objects for external MSAL components
+
+### Test Organization
+
+-   Follow Arrange-Act-Assert (AAA) pattern
+-   Descriptive test names explaining scenario
+-   Test all public methods and state transitions
+-   Cover all error scenarios and edge cases
+-   Test all error helper methods
+
+### Test Utilities
+
+-   Shared test utilities in `test_resources/` folder
+-   Reusable mock objects and test data
+-   Common test configuration patterns
+-   Helper functions for complex test scenarios
+
+### Integration Testing Requirements
+
+-   Integration tests located in `test/custom_auth/integration_tests/` folder
+-   Test end-to-end authentication flows with mocked API responses
+-   Validate complete user journeys across multiple components
+-   Test error scenarios and edge cases in integrated environments
+-   Mock API responses to simulate various authentication scenarios and error conditions
+
+## Performance Standards
+
+### Bundle Optimization
+
+-   Support tree shaking and dead code elimination
+-   Use dynamic imports for large/optional features
+-   Minimize method and property names for better minification
+-   Avoid patterns that prevent optimization
+
+### Runtime Efficiency
+
+-   Efficient state object creation and management
+-   Minimal memory allocation in hot paths
+-   Proper cleanup of resources and event listeners
+-   Optimized network request patterns
+
+## Development Workflow
+
+### Build Requirements
+
+All code must pass:
+
+```bash
+cd lib/msal-browser && npm run build:all
+cd lib/msal-browser && npm run format:fix
+cd lib/msal-browser && npm run lint
+```
+
+### Code Generation Patterns
+
+When adding new authentication flows:
+
+1. Add method to entry point and controller
+2. Create new interaction client
+3. Add state/result classes for each step
+4. Update state machine logic
+5. Add/extend network client methods
+6. Create comprehensive tests
+7. Update documentation
+
+### Dependency Management
+
+-   Request approval before adding new dependencies
+-   Minimize external dependencies
+-   Use existing MSAL Browser infrastructure where possible
```

Comment:
Same general comment here r.e. "minimize" and "existing" being too subjective and unspecific